components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt

```diff
@@ -178,7 +208,28 @@ class DownloadsFeature(
         ).show()
     }
 
-    private fun showDialog(tab: SessionState, download: DownloadState) {
+    @VisibleForTesting
+    internal fun showAppDownloaderDialog(tab: SessionState, download: DownloadState, apps: List<DownloaderApp>) {
+        appChooserDialog.setApps(apps)
+        appChooserDialog.onAppSelected = { app ->
+            if (app.packageName == applicationContext.packageName) {
+                startDownload(download)
+            } else {
+                applicationContext.startActivity(app.toIntent())
```

Comment:
Afaik Fennec also persists the choice (by mimetype?) - maybe something we could file and implement some other time if really needed?

Reply:
Exactly, I was a bit confused because Fennec stored the preferred app for handling a mimetype but I couldn't find an UI  to change your previous saved options. We can file UX ticket and ac.

Reply:
Heh, yeah, I think Fennec never exposed UI for that - which is indeed awkward.

Reply:
I filed https://github.com/mozilla-mobile/android-components/issues/7943 for it